src/Compatibility/Microsoft.DotNet.PackageValidation/ApiCompatRunner.cs

```diff
@@ -20,18 +18,30 @@ namespace Microsoft.DotNet.PackageValidation
     public class ApiCompatRunner
     {
         internal Dictionary<MetadataInformation, List<(MetadataInformation rightAssembly, string header)>> _dict = new();
-        private readonly ApiComparer _differ = new();
-        private readonly ICompatibilityLogger _log;
-        private readonly Dictionary<string, HashSet<string>> _referencePaths;
-        private bool _isBaselineSuppression = false;
-        private string _leftPackagePath;
-        private string _rightPackagePath;
-
-        public ApiCompatRunner(bool enableStrictMode, ICompatibilityLogger log, Dictionary<string, HashSet<string>> referencePaths)
+        private readonly ApiComparer _differ;
+        private readonly CompatibilityLoggerBase _log;
+        private readonly Dictionary<string, HashSet<string>>? _referencePaths;
+        private readonly string _leftPackagePath;
+        private readonly string _rightPackagePath;
+        private readonly bool _isBaselineSuppression;
+
+        public ApiCompatRunner(bool enableStrictMode,
+            CompatibilityLoggerBase log,
+            Dictionary<string, HashSet<string>>? referencePaths,
+            string leftPackagePath,
+            string? rightPackagePath = null)
         {
-            _differ.StrictMode = enableStrictMode;
+            _differ = new()
+            {
+                StrictMode = enableStrictMode
+            };
             _log = log;
-            _referencePaths = referencePaths ?? new();
+            _referencePaths = referencePaths;
+
+            _leftPackagePath = leftPackagePath;
+            _rightPackagePath = rightPackagePath ?? leftPackagePath;
+            // If assets from different packages are compared, mark the underlying suppression as baseline.
+            _isBaselineSuppression = !_leftPackagePath.Equals(_rightPackagePath, StringComparison.InvariantCultureIgnoreCase);
```

Comment:
Why not just make _rightPackagePath nullable and check here if it was null?

Reply:
The ApiCompatRunner relies on the `_rightPackagePath` variable in other places. Setting the right package path to the left package path makes it easier to later work with the strings. Also I didn't change that, it was already like this before.